tar/test/test_option_C_mtree.c

```diff
@@ -0,0 +1,68 @@
+/*-
+ * Copyright (c) 2018 The FreeBSD Foundation
+ * All rights reserved.
+ * 
+ * This software was developed by Arshan Khanifar <arshankhanifar@gmail.com>
+ * under sponsorship from the FreeBSD Foundation.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR(S) ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL THE AUTHOR(S) BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+#include "test.h"
+__FBSDID("$FreeBSD$");
+
+DEFINE_TEST(test_option_C_mtree)
+{
+	char *p0;
+	size_t s;
+	int r;
+
+	p0 = NULL;
+
+	char *content = "./foo type=file uname=root gname=root mode=0755\n";
+
+	/* Create an archive using an mtree file. */
+	assertMakeFile("/tmp/METALOG.mtree", 0777, content);
```

Comment:
This is tricky.  You can't rely on `/tmp` always existing on every platform that uses libarchive.  This is why libarchive's main test harness does so much work to set up a clean empty work directory for every test, so tests never need to hardcode things like this.

Most of the time, you can rely on the current working directory being suitable for your test to run.  Just use relative paths and all will be well.

For the rare cases where you _really_ _must_ have an absolute path, the global `testworkdir` variable holds the full absolute path of the test work directory.

Reply:
This bug occurs specifically in the case where an absolute path to an `mtree` file is given. Thus, I will need to use `testworkdir` to provide an absolute path as an argument.
Thanks, I'll update it with `testworkdir` + `/METALOG.mtree`, that way it should be safe.